lib/px/service/client/hmac_signing.rb

```diff
@@ -0,0 +1,38 @@
+module Px::Service::Client
+  module HmacSigning
+    extend ActiveSupport::Concern
+    
+    included do
+      alias_method_chain :make_request, :signing
+    end
+
+    def make_request_with_signing(method, uri, query: nil, headers: nil, body: nil)
+
+      signature = generate_signature(method, uri, query, body)
+      headers = {} if headers.nil?
+      headers.merge!("X-Service-Auth" => signature)
+
+      make_request_without_signing(
+          method,
+          uri,
+          query: query,
+          headers: headers,
+          body: body)
+    end
+
+    ##
+    # Generate a timestamp, a nonce and the corresponding HMAC signature
+    def generate_signature(method, uri, query, body)
+      keyspan = Px::Service::Client.config.keyspan
```

Comment:
This should read from the current class's config, not the global one.  Multiple clients can inherit from this class but won't share the same secrets.